vul-scans/action.yaml

```diff
@@ -0,0 +1,282 @@
+      run: |
+        ATTESTATION=$(mktemp)
+
+        export SCANNER_URI=$(cat trivy-results.sarif | jq .runs[0].tool.driver.informationUri)
+        export SCANNER_VERSION=$(cat trivy-results.sarif | jq .runs[0].tool.driver.version)
+
+        cat > "${ATTESTATION}" <<EOF
+        {
+            "invocation": {
+              "parameters": null,
+              "uri": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}",
+              "event_id": "${{ github.event.id }}",
+              "builder.id": "${{ github.workflow }}"
+            },
+            "scanner": {
+              "uri": ${SCANNER_URI},
+              "version": ${SCANNER_VERSION},
+              "result": $(cat trivy-results.sarif | jq .)
+            },
+            "metadata": {
+              "scanStartedOn": "${{ steps.scan-start.outputs.date }}",
+              "scanFinishedOn": "$(TZ=Zulu date "+%Y-%m-%dT%H:%M:%SZ")"
+            }
+        }
+        EOF 
+               
+        echo "trivy COSIGN_EXPERIMENTAL: $COSIGN_EXPERIMENTAL"
+        
+        export COSIGN_EXPERIMENTAL="$COSIGN_EXPERIMENTAL"
+               
+        cosign attest ${{ inputs.image }} \
+            --type vuln --predicate "${ATTESTATION}" \
+            --attachment-tag-prefix trivy-
+
+
+    - name: Scan image with Anchore/Grype
+      id: grype-scan
+      uses: anchore/scan-action@ecfd0e98932e57ea8f68f29c4f418fc41a8194db
+      with:
+        image: ${{ inputs.image }}
+        fail-build: false
+        severity-cutoff: low
+
+    - name: Cosign Attest Grype results
+      shell: bash
+      id: grype-attest
+      env:
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
+        ATTESTATION=$(mktemp)
+
+        export SCANNER_URI=$(cat ${{ steps.grype-scan.outputs.sarif }} | jq .runs[0].tool.driver.informationUri)
+        export SCANNER_VERSION=$(cat ${{ steps.grype-scan.outputs.sarif }} | jq .runs[0].tool.driver.version)
+
+        cat > "${ATTESTATION}" <<EOF
+        {
+            "invocation": {
+              "parameters": null,
+              "uri": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}",
+              "event_id": "${{ github.event.id }}",
+              "builder.id": "${{ github.workflow }}"
+            },
+            "scanner": {
+              "uri": ${SCANNER_URI},
+              "version": ${SCANNER_VERSION},
+              "result": $(cat ${{ steps.grype-scan.outputs.sarif }} | jq .)
+            },
+            "metadata": {
+              "scanStartedOn": "${{ steps.scan-start.outputs.date }}",
+              "scanFinishedOn": "$(TZ=Zulu date "+%Y-%m-%dT%H:%M:%SZ")"
+            }
+        }
+        EOF
+
+        echo "grype COSIGN_EXPERIMENTAL: $COSIGN_EXPERIMENTAL"
+        
+        export COSIGN_EXPERIMENTAL="$COSIGN_EXPERIMENTAL"
+          
+        cosign attest ${{ inputs.image }} \
+            --type vuln --predicate "${ATTESTATION}" \
+            --attachment-tag-prefix grype-
```

Comment:
I feel like this could be its own action to encapsulate things, since we keep copying it.

Reply:
I dont disagree, I was going to break out it.

Reply:
Starting testing it here https://github.com/strongjz/distroless-actions/tree/attest-action